### PR TITLE
Expand Native Asset Examples in Sway Book

### PR DESCRIPTION
**Pull request by SilentCicero**
- Added more native asset examples in the Sway Book
- Added context on what smart-contracts can do with native assets

Partially closes: https://github.com/FuelLabs/sway/issues/1281

### CODE REVIEW

File: docs/src/blockchain-development/native_assets.md [hunk 1]
```diff
@@ -7,3 +7,23 @@ What does this mean in practice?
 As in Ethereum, sending ETH to an address or contract is an operation built into the FuelVM, meaning it doesn't rely on the existence of some token smart contract to update balances to track ownership.
 
 However, unlike Ethereum, the process for sending _any_ native asset is the same. This means that while you would still need a smart contract to handle the minting and burning of fungible tokens, the sending and receiving of these tokens can be done independently of the token contract.
+
+## Liquidity Pool Example
+
+All contracts in Fuel can mint and burn their own native token. Contracts also manage an internal balance of all native assets pushed through calls or minted by the contract. Lastly, contracts can receive and transfer any native asset including their own.
```
Comment (nfurfaro) [inline docs/src/blockchain-development/native_assets.md:13]: Doesn't the protocol keep track of the balances of native assets?
The contract shouldn't need to do any accounting of it's own native assets.

Comment (nfurfaro) [reply docs/src/blockchain-development/native_assets.md:13]: That is, you can always query any contract's balance of any native asset.

Comment (SilentCicero) [reply docs/src/blockchain-development/native_assets.md:13]: Oh that's what I'm trying to refer to here. That the contract by virtue of the VM is tracking a balance.

Perhaps there is a clearer way to write that. Suggestions welcome.

File: docs/src/blockchain-development/native_assets.md [hunk 2]
```diff
@@ -7,3 +7,23 @@ What does this mean in practice?
...
+
+In this example, we show a basic liquidity pool contract minting its own native asset LP token.
+
+```sway
+{{#include ../../../examples/liquidity_pool/src/main.sw}}
+```
+
+## Native Token Example
+
+In this example, we show a native token with more minting, burning and transfering capibility.
```
Comment (mohammadfawaz) [inline docs/src/blockchain-development/native_assets.md:25]:
```suggestion
In this example, we show a native token contract with more minting, burning and transferring capabilities.
```

File: examples/liquidity_pool/src/main.sw [hunk 3]
```diff
@@ -0,0 +1,42 @@
+contract;
+
+use std::{
+    address::Address,
+    context::msg_amount,
+    context::call_frames::{msg_asset_id, contract_id},
+    assert::assert,
+    contract_id::ContractId,
+    token::{mint_to_address, transfer_to_output}
+};
```
Comment (mohammadfawaz) [inline examples/liquidity_pool/src/main.sw:10]: I like these ordered alphabetically. I think the formatter does that but it won't nicely break the whole `use` statement like this. Relevant: https://github.com/FuelLabs/sway/issues/1209.

**Review by mohammadfawaz — CHANGES_REQUESTED** [review]

File: examples/liquidity_pool/src/main.sw [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+abi LiquidityPool {
+    fn deposit(recipient: Address);
+    fn withdraw(recipient: Address);
+}
+
+const BASE_TOKEN:b256 = 0x9ae5b658754e096e4d681c548daf46354495a437cc61492599e33fc64dcdc30c;
```
Comment (adlerjohn) [inline examples/liquidity_pool/src/main.sw:17]: format sway files

File: examples/native_token/src/main.sw [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    fn force_transfer_coins(coins: u64, asset_id: ContractId, target: ContractId);
+    fn transfer_coins_to_output(coins: u64, asset_id: ContractId, recipient: Address);
+    fn deposit();
+    fn get_balance(target: ContractId, asset_id: ContractId) -> u64;
+    fn mint_and_send_to_contract(amount: u64, destination: ContractId);
+    fn mint_and_send_to_address(amount: u64, recipient: Address);
+}
+
+impl NativeAssetToken for Contract {
+    // Mint an amount of this contracts native asset to the contracts balance.
```
Comment (adlerjohn) [inline examples/native_token/src/main.sw:23]: Use `///` for docstrings.

File: docs/src/blockchain-development/native_assets.md (hunk 1 above)
Comment (nfurfaro) [reply docs/src/blockchain-development/native_assets.md:13]: I think that's better, with a small tweak:
"Therefore, there is no need for any manual accounting of the contract's balances using persistent storage."

File: examples/liquidity_pool/src/main.sw (hunk 4 above)
Comment (SilentCicero) [reply examples/liquidity_pool/src/main.sw:17]: I did, and still got the same result on these lines @adlerjohn

Comment (SilentCicero) [reply examples/liquidity_pool/src/main.sw:17]: Ah, I see now, just a space between the colon and type. Got it.

**Review by adlerjohn — APPROVED** [review]

**Review by nfurfaro — APPROVED** [review]

**Review by mohammadfawaz — APPROVED** [review]
LGTM!

**Pull request merged**